### DISCUSSION TITLE
Introducing VanGraph: Visualizing Dependencies in Your App

### DISCUSSION THREAD

**Discussion by Tao-VanJS** [vanjs-org/van#361] [Announcements]
Hi fellow VanJSers,

I'm thrilled ot announce: **VanGraph**, a library that helps you visualize dependencies among states and DOM nodes in your app with the help of [Graphviz](https://graphviz.org/). Here is the sample usage:

```js
const firstName = van.state("Tao"), lastName = van.state("Xin")
const fullName = van.derive(() => `${firstName.val} ${lastName.val}`)

// Build the DOM tree...

// To visualize the dependency graph among `firstName`, `lastName`, `fullName`, and all the
// derived states and DOM nodes from them.
vanGraph.show({firstName, lastName, fullName})
```

Checkout a demo in [CodeSandbox](https://codesandbox.io/p/devbox/github/vanjs-org/van/tree/main/graph/examples/basic?file=%2Fsrc%2Fmain.ts).

For more information about its usage, you can checkout its [home page](https://github.com/vanjs-org/van/tree/main/graph).

**Comment by kangaroolab** [comment]
very cool...do you have example how to use it in browser?

**Reply by Tao-VanJS** [reply]
I think there is a CodeSandbox example in this thread.
